testsuite/smoke-test/src/aptos/move_modules/sources/hello.move

```diff
@@ -2,6 +2,18 @@ module 0xA550C18::HelloWorld {
     use AptosFramework::Coin;
     use AptosFramework::TestCoin::TestCoin;
 
+    struct ModuleData has key, store {
+        global_counter: u64,
+    }
+
+    fun init_module(creator: &signer) {
+        move_to(
+            creator,
+            ModuleData {
+                global_counter: 0}
```

Comment:
nit
```suggestion
                global_counter: 0,
            }
```
```suggestion
                global_counter: 0}
```

Reply:
Usecases are: 
1. Initializing global variables with complex types. Currently, we only support initializing a const primitive type
For example, default config for token and collect mutability

2. Intializing a global counter/id. 
For example, maintain a variable called id and initalized it as 0 when the module publishes. For example, proposal_id.

Reply:
actually once we land this, we can swap our coin model to be a bit more move native and actually support some concept of write-once autonomous accounts.